### PR TITLE
Add check for text-to-speech

### PR DESCRIPTION
**Pull request by owades**
# Description

This check assesses schedule feed compliance to the guideline:

> The tts_stop_name field should include correct pronunciation for all stop names in stops.txt that are commonly mispronounced in trip planning applications.

The check is performed by assessing whether common text-to-speech needs are missed. If any of the following searches are met, AND the tts_stop_name is null (or it is identical to stop_name), the test will FAIL. Otherwise it will pass.

The implemented searches are:
- Check for existence of any of the following strings, standing on their own (ie separated by a space):  "n","s","e","w","ne","se","sw","nw","nb","sb","eb","wb","st","rd","blvd","hwy"
- Check for >=2 adjacent numerals ("21" should read "twenty one")
- Check for presence of any of the following symbols: "/", "(", ")" ("pine/baker" should read "pine and baker")

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?
Manually audit 5 passing and 5 failing results, to confirm why each one passed and failed.
Separate out each of the text rules, and confirm that it captures stop_names matching those rules
Ran `dbt test`

### CODE REVIEW

**Comment by owades** [comment]
@lauriemerrell, could you advise on how to resolve the lint test issues?

File: warehouse/models/mart/gtfs_guidelines/guideline_checks.md [hunk 1]
```diff
@@ -24,4 +24,5 @@ Here is a list of currently-implemented checks:
...
 | Service alerts RT feed is listed on feed aggregator transit.land | Feed Aggregator Availability | Service alerts RT feed is listed on feed aggregator transit.land on the given day. Due to current limitations, this check will be less reliable for agencies with multiple service alerts feeds listed in agencies.yml. If one of the service alerts feeds listed is on transit.land, it will pass for all feeds. |
 | Every trip in trips.txt has a shape_id listed | Accurate Service Data | Every row in trips.txt has a non-null value in the shape_id column. |
+| Include tts_stop_name entries in stops.txt for stop names that are pronounced incorrectly in most mobile applications. | Accurate Accessibility Data | For every stop_name in stops.txt containing a commonly mispronounced word, there is a non-null tts_stop_name field. |
```
Comment (lauriemerrell) [inline warehouse/models/mart/gtfs_guidelines/guideline_checks.md:27]: Worth mentioning here that it also checks for whether it's identical to `stop_name`?

**Review by lauriemerrell — APPROVED** [review]
One optional docs suggestion. Linter looks ok now? @atvaccaro also updating some precommit versions in #1950 because he ran into issues too.

File: warehouse/models/mart/gtfs_guidelines/guideline_checks.md (hunk 1 above)
Comment (owades) [reply warehouse/models/mart/gtfs_guidelines/guideline_checks.md]: Thanks! I included this in an overhaul of the description. Might as well be very specific.

**Pull request merged**